src/Elastic.Apm/DiagnosticListeners/HttpDiagnosticListenerImplBase.cs

```diff
@@ -97,7 +99,7 @@ public void OnNext(KeyValuePair<string, object> kv)
 
 			if (IsRequestFilteredOut(requestUrl))
 			{
-				Logger.Trace()?.Log("Request URL ({RequestUrl}) is filtered out - exiting", requestUrl);
+				Logger.Trace()?.Log("Request URL ({RequestUrl}) is filtered out - exiting", Http.Sanitize(requestUrl));
```

Comment:
`Http.Sanitize` is going to be called regardless of whether trace logging is enabled, right? That's going to add some overhead. Maybe worth adding some kind of lazy evaluation for logging?

Reply:
@SergeyKleyman clued me in - I thought the arguments would still be evaluated, it just wouldn't call Log with them, but that is not the case.